### DISCUSSION TITLE
Tool input parsing error with Claude via AWS Bedrock in Agents endpoint

### DISCUSSION THREAD

**Discussion by ogipogi** [danny-avila/LibreChat#4638] [Troubleshooting] [answered]
### What happened?

When using the Agents endpoint with Claude via AWS Bedrock and a custom RAG tool, the tool input parsing fails. The error occurs because the input is a plain string, not a JSON object as expected.

Root Cause: The JSON.parse(args) is failing because the string "How to write a unit test in XY? Include best practices and examples." is not a valid JSON string.

https://github.com/danny-avila/LibreChat/blob/main/api/app/clients/prompts/formatMessages.js#L191-L195 

Possible Fix: Modify the code to handle different input types:
```
let args = _args;
if (typeof _args === 'string') {
  args = { input: _args };
} else {
  try {
    args = JSON.parse(_args);
  } catch (e) {
    args = { input: _args };
  }
}
```

If you want, I can create a PR?

Thanks for checking.

### Steps to Reproduce

1. Set up LibreChat with Agents endpoint using Claude via AWS Bedrock
2. Attach a custom RAG tool to the agent
3. Use the RAG tool during a conversation
a) How implement a unit test in [PRODUCT-NAME]
b) And how write a unit test for a resolver in [PRODUCT-NAME]?
5. Observe the error in the logs

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
SyntaxError: Unexpected token H in JSON at position 0
at JSON.parse (<anonymous>)
at formatAgentMessages (api/app/clients/prompts/formatMessages.js:192:23)
at AgentClient.chatCompletion (api/server/controllers/agents/client.js:482:24)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I wrote a note for this that needing to do this might be provider specific. After some testing, other endpoints don't seem to be negatively affected by a potential fix, however json parsing should be attempted first, as a valid json string should be used outright

**Reply by danny-avila** [reply, marked as answer]
patching here: https://github.com/danny-avila/LibreChat/pull/4641

**Reply by ogipogi** [reply]
That makes sense 👍 Thanks for the patch!
